db/silkworm/db/bitmap.cpp

```diff
@@ -68,4 +68,38 @@ roaring::Roaring64Map cut_left(roaring::Roaring64Map &bm, uint64_t size_limit) {
     return res;
 }
 
+roaring::Roaring cut_left(roaring::Roaring &bm, uint64_t size_limit) {
```

Comment:
Do not copy and paste code! That's a bad practice.

Use C++ function template instead if `roaring::Roaring64Map` &  `roaring::Roaring` don't have a common superclass. Something like
```
template <typename RoaringMapType>
RoaringMapType cut_left(RoaringMapType& bm, uint64_t size_limit) {
    if (bm.getSizeInBytes() <= size_limit) {
...
```

You probably have to move the function to the header if it's a template.

Reply:
Oh, I see.